py-polars/src/conversion.rs

```diff
@@ -610,6 +694,24 @@ impl<'s> FromPyObject<'s> for Wrap<AnyValue<'s>> {
                     let v = time.extract::<i64>().unwrap();
                     Ok(Wrap(AnyValue::Time(v)))
                 }),
+                "Decimal" => {
+                    // note: there seems to be no way to extract precision from decimal.Decimal
```

Comment:
Correct, though I believe it's available using `getcontext()` which is a thread local. I'm not sure it's a good idea to call `getcontext()` here, but I believe you can extract precision like this (more or less):

```rust
let py_decimal = PyModule::import(py, "decimal")?;
let ctx = py_decimal.call_method("getcontext", (), None)?;
let precision = ctx.getattr("prec");
```

Reply:
Nope, IIUC `getcontext()` returns you the current context used to build *_new_* decimals. It doesn't tell you anything about existing ones.

```python
print(decimal.Context(prec=2).create_decimal('1.23456'))
print(decimal.getcontext().prec)
```
```
1.2
28
```

Reply:
I see. I suppose that makes sense since it's only relevant when computing over decimals; in which case, you need `Context` for creating new decimals and rounding and such. Even more argument towards dropping `precision`!

That said, if we want *some* value, using the thread's `Context` might still be something to consider: it is the "most likely" value of `precision` and the behaviour wouldn't be surprising unless you're doing something very specific.

Reply:
See some of my thoughts above, tbh I'd be inclined towards completely dropping `precision` from `AnyValue` as it's basically unused now and potentially complicates "from any-values" procedure by an order of magnitude if it's to be respected.